File: android/app/src/main/java/com/scascan/app/domain/usecase/SearchFoodUseCase.kt
package com.scascan.app.domain.usecase

import com.scascan.app.data.model.NutritionFacts
import com.scascan.app.data.repository.NutritionRepository
import javax.inject.Inject

class SearchFoodUseCase @Inject constructor(
    private val repository: NutritionRepository
) {
    suspend operator fun invoke(query: String): Result<NutritionFacts> =
        repository.searchFood(query)
}

File: ios/ScaScanKit/Sources/ScaScanKit/Domain/UseCases/SearchFoodUseCase.swift
/// Mirrors Android's `domain.usecase.SearchFoodUseCase`.
public struct SearchFoodUseCase: Sendable {
    private let repository: NutritionRepository

    public init(repository: NutritionRepository) {
        self.repository = repository
    }

    public func callAsFunction(_ query: String) async throws -> NutritionFacts {
        try await repository.searchFood(query)
    }
}
